dockstore-client/src/main/java/io/dockstore/client/cli/nested/ApiClientExtended.java

```diff
@@ -16,16 +17,56 @@
 
 public class ApiClientExtended extends ApiClient {
 
+    /**
+     *
+     * @param key
+     * @return
+     */
+    private MediaType getMediaType(String key) {
```

Comment:
1. I would return an `Optional<MediaType>`.  If a method can return null, then it should return an Optional instead. You end up putting the return value into an Optional anyway, on line 53, so you might as well make it return an Optional here.
2. The parameter `key` should probably  be an `Optional<String>`, which I think is preferred. If you'd rather not do that, I feel you should just do a null check, instead of creating an `Optional` on the next line for the only purpose of calling `isPresent` 3 lines later.

Reply:
implemented the suggestions